### PR TITLE
Cleanup & add compiler warning options.

### PR DESCRIPTION
**Pull request by ArthurSonzogni**
- Add compiler warning. This avoids common problems.

- Add a top-level CMakeLists. This allows developers to know they can
  use CMake and can directly use cmake using `cmake ..` from the build/
  directory, instead of cmake `../board`.

- Merge the few Makefile into a shorter one.

- Remove the bin directory. Output directly into the build directory.

- Add tools/format.sh to run clang-format on every files.

- Format every files.

### CODE REVIEW

**Comment by SAtacker** [comment]
Beagle-config has a host part that might be using Qt. That's why I had it under the board directory. I'll have a new repository for the host if that's bettter.

**Comment by ArthurSonzogni** [comment]
Yes. And we can move the content of src/ into src/board when you will add the src/host.
I can rename if src/ board/ if you really care about this. Please let me know.

**Comment by SAtacker** [comment]
I'll do it later and add an option in CMakeLists.txt

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  PRIVATE "-pedantic"
+  PRIVATE "-Werror"
+  PRIVATE "-Wdeprecated"
+  PRIVATE "-Wshadow"
+)
+
+# Useful for debugging and find 
+if (BEAGLE_CONFIG_SANITIZE)
+  set(CMAKE_C_COMPILER clang)
+  set(CMAKE_CXX_COMPILER clang++)
```
Comment (SAtacker) [inline CMakeLists.txt:56]: I would like to retain the compilers given from the command line as CMAKE parameters.

**Pull request closed without merging**